resources/js/store.ts

```diff
@@ -62,15 +63,42 @@ export class Store {
     showLoadingOverlay();
     showLoadingOverlay.flush();
 
-    let checkout: any;
-    try {
-      // create shopify checkout.
-      // error returned will be a JSON string in error.message
-      checkout = await client().checkout.create({
-        customAttributes: [{ key: 'orderId', value: orderId }],
-        lineItems: this.collectShopifyItems(),
-      });
-    } catch (_error) {
+    const operation = `
+      mutation CreateCart($cartInput: CartInput) {
+        cartCreate(input: $cartInput) {
+          cart {
+            id
+            checkoutUrl
+            lines(first: 10) {
+              edges {
+                node {
+                  id
+                  merchandise {
+                    ... on ProductVariant {
+                      id
+                      title
+                    }
+                  }
+                }
+              }
+            }
+            cost {
+              totalAmount {
+                amount
+                currencyCode
+              }
+            }
+          }
+        }
+      }
+    `;
+
+    // create shopify checkout.
+    // error returned will be a JSON string in error.message
+    const response = await storefrontClient().request(operation, { variables: this.shopifyCartInput(orderId) });
+    const data = response.data as { cartCreate: CartCreatePayload };
```

Comment:
casting the property of the response because I'm not about to go through the whole graphql type generation process for 2 queries right now.

Reply:
you mean like every single json api responses here? 👀